expr/func_movingwindow.go

```diff
@@ -0,0 +1,177 @@
+package expr
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/grafana/metrictank/batch"
+	"github.com/grafana/metrictank/errors"
+	"github.com/grafana/metrictank/schema"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/consolidation"
+	"github.com/raintank/dur"
+)
+
+type FuncMovingWindow struct {
+	in           GraphiteFunc
+	generic      bool
+	windowSize   string
+	fn           string
+	xFilesFactor float64
+
+	shiftOffset uint32
+}
+
+// NewMovingWindowParticular constructs a known, particular function (movingAverage, movingMax, etc)
+func NewMovingWindowParticular(name string) func() GraphiteFunc {
+	return func() GraphiteFunc {
+		return &FuncMovingWindow{fn: name}
+	}
+}
+
+// NewMovingWindowGeneric constructs a generic movingWindow function
+func NewMovingWindowGeneric() GraphiteFunc {
+	return &FuncMovingWindow{
+		fn:      "average",
+		generic: true,
+	}
+}
+
+func (s *FuncMovingWindow) Signature() ([]Arg, []Arg) {
+	if s.fn == "" {
```

Comment:
```suggestion
	if s.generic {
```
?

Reply:
heh can't believe i missed this :)

Reply:
fixed https://github.com/grafana/metrictank/pull/1941/commits/65b9f0c36f77f7c143836160101950c8a01f0811